workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Readers/ReaderBase.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using Improbable.Worker;
+using Unity.Entities;
+using UnityEngine;
+using Entity = Unity.Entities.Entity;
+
+namespace Improbable.Gdk.Core
+{
+    public abstract class ReaderBase<TSpatialComponentData, TComponentUpdate>
+        : IReader<TSpatialComponentData>,
+            IReaderInternal
+        where TSpatialComponentData : ISpatialComponentData
+        where TComponentUpdate : ISpatialComponentUpdate<TSpatialComponentData>
+    {
+        protected readonly Entity Entity;
+        protected readonly EntityManager EntityManager;
+
+        protected ReaderBase(Entity entity, EntityManager entityManager)
+        {
+            Entity = entity;
+            EntityManager = entityManager;
+        }
+
+        public Authority Authority
+        {
+            get
+            {
+                if (EntityManager.HasComponent<Authoritative<TSpatialComponentData>>(Entity))
+                {
+                    return Authority.Authoritative;
+                }
+
+                if (EntityManager.HasComponent<AuthorityLossImminent<TSpatialComponentData>>(Entity))
+                {
+                    return Authority.AuthorityLossImminent;
+                }
+
+                if (EntityManager.HasComponent<NotAuthoritative<TSpatialComponentData>>(Entity))
+                {
+                    return Authority.NotAuthoritative;
+                }
+
+                throw new Exception("No authority component found for this entity.");
```

Comment:
What happened? How can the user fix it?

Reply:
They can't - its a failing in the core somewhere

Reply:
So far, we've been creating bespoke exception types for each exception. Please make an AuthorityTagNotFoundException sort of custom exception class.